config/api_config.go

```diff
@@ -55,8 +55,8 @@ func (cfg *APIConfig) SetDefaults() {
 }
 
 // URL provides the API URL for a given endpoint key.
-func (cfg *APIConfig) URL(key string) string {
-	return fmt.Sprintf("%s%s", cfg.BaseURL, cfg.Endpoints[key])
+func (cfg *APIConfig) URL(key string, args ...interface{}) string {
+	return fmt.Sprintf(fmt.Sprintf("%s%s", cfg.BaseURL, cfg.Endpoints[key]), args...)
```

Comment:
Nice refactor- but the double Sprintf threw me a bit. Maybe split this into two lines?